### DISCUSSION TITLE
Hyperformula - release memory

### DISCUSSION THREAD

**Discussion by MichaelHilgers** [handsontable/hyperformula#928] [General]
How can i release memory of hyperformula object without destroing the object?
I use the hf object and read all values with the getAllSheetsValues() function.
If i do this many times, the memory grov up, if i wait after this 1min then memory go down to the initial size.
How i can free memory manuelly ?

Thanks

**Comment by sequba** [comment]
Hi @MichaelHilgers! 

Unfortunately there is no easy way to free memory manually. HF has no custom memory management.

> If i run getAllSheetsValues() many times, the memory grov up, if i wait after this 1min then memory go down to the initial size.

The behavior you observed is caused by standard javascript garbage collector. Some javascript engines may expose the garbage collector's API so that you could run it manually but it is vendor-specific and generally not recommended. Here is an idea of how GC can be run manually in node-based environments: https://stackoverflow.com/a/30654451

If HF uses too much memory for your use-case, please report an issue and include your code and sheet data so that we could investigate it.
